packages/teams-js/src/public/interfaces.ts

```diff
@@ -169,6 +169,75 @@ export enum FileOpenPreference {
   Web = 'web',
 }
 
+/**
+ * Possible Action Types
+ */
+export enum ActionObjectType {
+  M365Content = 'm365content',
+}
+
+/**
+ * Data pertaining to object(s) the action is being performed on
+ *
+ * @param T The type of action being implemented
+ */
+export interface BaseActionObject<T extends ActionObjectType> {
+  type: T;
+}
+
+/**
+ * Stores information needed to represent M365 Content stored
+ * in OneDrive or Sharepoint
+ */
+export interface M365ContentAction extends BaseActionObject<ActionObjectType.M365Content> {
+  /**
+   * Only office content IDs are passed to the app. Apps should use these ids
+   * to query the Microsoft graph for more details.
+   */
+  itemId: string;
+  secondaryId?: SecondaryId;
+}
+
+/**
+ * Contains information on what Graph item is being queried
+ */
+export interface SecondaryId {
+  name: SecondaryM365ContentIdName;
+  value: string;
+}
+
+/**
+ * These correspond with field names in the MSGraph
+ */
+export enum SecondaryM365ContentIdName {
+  DriveId = 'driveId',
+  GroupId = 'groupId',
+  SiteId = 'siteId',
+  UserId = 'userId',
+}
+
+/**
+ * Type guard to determine if an action item is of M365Content Type
+ */
+export function isM365ContentType(actionItem: unknown): actionItem is M365ContentAction {
```

Comment:
This probably isn't the right place to put this since this file is for interfaces/enums and not functions. It also looks like this is only used in the test, so maybe more appropriate to put in the test file

Reply:
(We also wouldn't add public facing documentation for test only stuff no matter what file it's in)

Reply:
sure I will add it to the test, I put it there because I didn't know if we would be expanding this later.  Since we only have one type of content action I figured that this could be used later